instagram/src/App.js

```diff
@@ -1,24 +1,71 @@
 import React, { Component } from 'react';
 import './App.css';
-import dummyData from "./dummy-data";
-import SearchBar from "./Components/SearchBar/SearchBar";
-import PostContainer from "./Components/PostContainer/PostContainer";
+import SearchBar from './Components/SearchBar/SearchBar';
+import PostContainer from './Components/PostContainer/PostContainer';
+import postData from './dummy-data.js';
 
 class App extends Component {
-  constructor() {
+
+  constructor(props) {
     super();
+    
+    this.state = {
+      allPosts: [],
+      displayedPosts: []
+    }
+
+    this.filterPosts = this.filterPosts.bind(this);
+  }
+
+ 
+
+  componentDidMount() {
+    this.setState({
+      allPosts: postData,
+      displayedPosts: postData
+    });
   }
 
+  filterPosts(criterion) {
+    if (criterion === '') {
+      this.setState({
+        displayedPosts: this.state.allPosts
+      });
+    } else {
+      const filteredPosts = this.state.allPosts.filter(post => post.username.includes(criterion));
+      this.setState({
+        displayedPosts: filteredPosts
+      });
+    }
+  }
+
```

Comment:
Nice clean code here. You can also put `filteredPosts` as a variable and compute the value during the rendering before `return`, this way, you don't need have two states for the posts.